src/backend/distributed/executor/intermediate_results.c

```diff
@@ -596,26 +596,33 @@ CreateIntermediateResultsDirectory(void)
 {
 	char *resultDirectory = IntermediateResultsDirectory();
 
-	if (!CreatedResultsDirectory)
+	MemoryContext intermediateResultsContext = AllocSetContextCreateExtended(
```

Comment:
not sure this is really useful unless we store it somewhere and reset it separately, maybe just use TopTransactionContext.